application/ui/components/params.jsx

```diff
@@ -0,0 +1,189 @@
+/** @jsx React.DOM */
+'use strict';
+
+var _               = require('underscore');
+var React           = require('react');
+var ParamRow        = require('./param-row');
+
+module.exports = React.createClass({
+    displayName : 'Params',
+
+    propTypes : {
+        params   : React.PropTypes.array,
+        onChange : React.PropTypes.func
+    },
+
+    getInitialState : function()
+    {
+        var values   = {};
+        var includes = {};
+
+        _.each(this.props.params, function(param) {
+            includes[param.name] = true;
+
+            if (typeof param.defaultValue !== 'undefined') {
+                values[param.name] = param.defaultValue;
+                return;
+            }
+
+            switch (param.type) {
+                case 'string':
+                    values[param.name] = param.defaultValue ? param.defaultValue : '';
```

Comment:
You may already know this, but in JavaScript there's a sort of "ternary shorthand" you could use instead for these:
```JS
param.defaultValue || ''
```

It's equivalent to the PHP,
```PHP
param->defaultValue ?: ''
```